pkg/build/controller/factory/factory.go

```diff
@@ -261,6 +260,45 @@ func (factory *ImageChangeControllerFactory) Create() controller.RunnableControl
 	}
 }
 
+type BuildConfigControllerFactory struct {
+	Client                  osclient.Interface
+	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	// Stop may be set to allow controllers created by this factory to be terminated.
+	Stop <-chan struct{}
+}
+
+// Create creates a new ConfigChangeController which is used to trigger builds on creation
+func (factory *BuildConfigControllerFactory) Create() controller.RunnableController {
+	queue := cache.NewFIFO(cache.MetaNamespaceKeyFunc)
+	cache.NewReflector(&buildConfigLW{client: factory.Client}, &buildapi.BuildConfig{}, queue, 2*time.Minute).RunUntil(factory.Stop)
+
+	bcController := &buildcontroller.BuildConfigController{
+		BuildConfigInstantiator: factory.BuildConfigInstantiator,
+	}
+
+	return &controller.RetryController{
+		Queue: queue,
+		RetryManager: controller.NewQueueRetryManager(
+			queue,
+			cache.MetaNamespaceKeyFunc,
+			func(obj interface{}, err error, retries controller.Retry) bool {
+				bc := obj.(*buildapi.BuildConfig)
+				if maxRetries > retries.Count {
```

Comment:
I might be missing something, but should this condition be the other way around `if retries.Count > maxRetries`. Unless I'm missing something `maxRetries` will be bigger than `retries.Count` from begining, which eventually won't retry those 60 times, isn't it?